feature-libs/product/variants/core/guards/product-variants.guard.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  UrlTree,
+} from '@angular/router';
+import {
+  Product,
+  ProductScope,
+  ProductService,
+  SemanticPathService,
+  VariantOption,
+} from '@spartacus/core';
+import { Observable, of } from 'rxjs';
+import { filter, map, switchMap, take } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProductVariantsGuard implements CanActivate {
```

Comment:
I'm thinking about keeping it in the components entry point (as before) so core won't be basically needed for now?

Reply:
Yes, basically (for now) Core was created just to handle guard. Not sure if I like the idea of breaking the concept that we have for future libs, but it's up to discussion.